### ISSUE TITLE
The replicaset error contradicts the service endopoint when I put the webhook and controller code together

### ISSUE THREAD

**Issue by zerunhu** [kubernetes-sigs/kubebuilder#3052]
### What broke? What's expected?

After I wrote the controller and crd code using kubebuilder, I wanted to add a webhook for core type to the code, so I followed the documentation directly from `5.5.2 Admission Webhook for Core Types`, and followed `1.9.2 Deploying Admission Webhooks` to deploy and all went well.
I need to do some tests, so i modified the controller's deployment's replicas from 1 -> 0 and then modified him from 0 -> 1, but the pod will not be created, the deployment does not have any errors, and when I `kubectl describe replicaset`, I find the following error
```
Events:
  Type     Reason            Age                From                   Message
  ----     ------            ----               ----                   -------
  Normal   SuccessfulCreate  4m43s              replicaset-controller  Created pod: tcaplus-controller-manager-654b78675c-zbnbl
  Normal   SuccessfulDelete  50s                replicaset-controller  Deleted pod: tcaplus-controller-manager-654b78675c-zbnbl
  Warning  FailedCreate      3s (x13 over 23s)  replicaset-controller  Error creating: Internal error occurred: failed calling webhook "mpod.kb.io": Post "https://tcaplus-webhook-service.tcaplus-system.svc:443/mutate-v1-pod?timeout=10s": no endpoints available for service "tcaplus-webhook-service"
```
The error is obvious: the creation of this pod requires access to a service called "tcaplus-webhook-service", but it does not have an endpoint so it cannot be accessed, I checked this service via `kubectl describe svc tcaplus-webhook-service`and found the endpoint to be empty
The endpoint of the service needs to be created by the controller's pod, but the pod cannot be created because the service does not have an endpoint, is this a contradiction?

### KubeBuilder (CLI) Version

3.2.0

### PROJECT version

version: "3"

### Other versions

`
go version：1.16
controller-runtime：v0.10.0

kubenetes version：1.18.4
`

**Comment by camilamacedo86** [comment]
Hi @zerunhu, 

Sorry but it shows hard to follow up on your scenario here. WDYT about we provide the detailed steps to reproduce your issue? So, could you try to do the same from the scratch using the latest version and let us know:

- a) What are the steps that we need to follow up to reproduce the scenario? Could you describe all so that we can create a project from the scratch following your instructions to check it out?
- b) What is the behaviour faced? What are the commands that did you use to deploy the project? Are you following the quick start to test it out? are you running on the cluster or are you using make run?
- c) Did you enable cert-manager as well? If so, how are you configuring it?
- d) What is the behaviour faced?
- e) What is the expected behaviour?

**Comment by zerunhu** [comment]
fix in https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#avoiding-deadlocks-in-self-hosted-webhooks
